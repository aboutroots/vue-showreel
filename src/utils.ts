export const sleep = async (ms: number): Promise<void> => {
  /**
   * Sleep for given number of miliseconds
   */
  // eslint-disable-next-line no-promise-executor-return
  await new Promise((r) => setTimeout(r, ms));
};
