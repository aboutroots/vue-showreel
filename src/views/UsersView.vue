<template>
  <section class="UsersView container">
    <h1>Users</h1>
    <p>Click on user card to view user profile.</p>
    <Loader v-if="processing" />
    <div v-else class="cards-wrapper fadeInBasic">
      <UserCard v-for="user in users" :key="user.id" :user="user" />
    </div>
  </section>
</template>

<script lang="ts">
import UserCard from '@/components/UserCard.vue';
import Loader from '@/components/SimpleLoader.vue';

import { mapGetters, mapActions } from 'vuex';

import Vue from 'vue';

interface State {
  processing: boolean;
}

export default Vue.extend({
  name: 'UsersView',
  components: {
    UserCard,
    Loader,
  },
  data() {
    return {
      processing: false,
    } as State;
  },

  computed: {
    ...mapGetters('users', {
      users: 'usersList',
    }),
  },

  async mounted(): Promise<void> {
    if (this.users.length === 0) {
      await this.loadUsers();
    }
  },

  methods: {
    ...mapActions('users', ['FETCH_USERS']),

    async loadUsers(): Promise<void> {
      this.processing = true;
      const success = await this.FETCH_USERS();
      if (!success) {
        this.$toast.error('Some error occured!');
      }
      this.processing = false;
    },
  },
});
</script>

<style lang="scss" scoped>
.cards-wrapper {
  display: flex;
  flex-wrap: wrap;
}
</style>
