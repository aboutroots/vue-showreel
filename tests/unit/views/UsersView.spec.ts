import flushPromises from 'flush-promises';
import UsersView from '@/views/UsersView.vue';

import { shallowMount } from '../utils';

describe('UsersView tests suite', () => {
  test('renders the component successfully', async () => {
    const wrapper = shallowMount(UsersView);
    await flushPromises();

    const container = wrapper.find('.UsersView');

    expect(container.exists()).toBeTruthy();
  });

  test('matches snapshot', async () => {
    const wrapper = shallowMount(UsersView);
    await flushPromises();

    expect(wrapper.element).toMatchSnapshot();
  });
});
