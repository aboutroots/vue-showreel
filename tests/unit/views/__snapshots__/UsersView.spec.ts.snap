// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersView tests suite matches snapshot 1`] = `
<section
  class="UsersView container"
>
  <h1>
    Users
  </h1>
   
  <p>
    Click on user card to view user profile.
  </p>
   
  <div
    class="cards-wrapper fadeInBasic"
  />
</section>
`;
